chains/evm/contracts/pools/USDC/USDCTokenPoolProxy.sol

```diff
@@ -0,0 +1,378 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.24;
+
+import {IPoolV1} from "../../interfaces/IPool.sol";
+import {IRouter} from "../../interfaces/IRouter.sol";
+import {ITypeAndVersion} from "@chainlink/contracts/src/v0.8/shared/interfaces/ITypeAndVersion.sol";
+
+import {ERC165CheckerReverting} from "../../libraries/ERC165CheckerReverting.sol";
+import {Pool} from "../../libraries/Pool.sol";
+import {USDCSourcePoolDataCodec} from "../../libraries/USDCSourcePoolDataCodec.sol";
+import {USDCTokenPool} from "./USDCTokenPool.sol";
+
+import {Ownable2StepMsgSender} from "@chainlink/contracts/src/v0.8/shared/access/Ownable2StepMsgSender.sol";
+import {IERC20} from "@openzeppelin/contracts@4.8.3/token/ERC20/IERC20.sol";
+import {SafeERC20} from "@openzeppelin/contracts@4.8.3/token/ERC20/utils/SafeERC20.sol";
+import {IERC165} from "@openzeppelin/contracts@4.8.3/utils/introspection/IERC165.sol";
+
+/// @dev The flag used to indicate that the source pool data is coming from a chain that does not have CCTP Support,
+/// and so the lock release pool should be used. The BurnMintWithLockReleaseTokenPool uses this flag as its source pool
+/// data to indicate that the tokens should be released from the lock release pool rather than attempting to be minted
+/// through CCTP.
+/// @dev The preimage is bytes4(keccak256("NO_CCTP_USE_LOCK_RELEASE")).
+bytes4 constant LOCK_RELEASE_FLAG = 0xfa7c07de;
```

Comment:
Probably should also move to the lib and live with the other tags that should be hard coded